Lib/test/libregrtest/runtest.py

```diff
@@ -434,10 +434,6 @@ def regrtest_runner(result, test_func, ns) -> None:
 def _load_run_test(result: TestResult, ns: Namespace) -> None:
     # Load the test function, run the test function.
     module_name = abs_module_name(result.test_name, ns.testdir)
-
-    # Remove the module from sys.module to reload it if it was already imported
-    sys.modules.pop(module_name, None)
```

Comment:
Hum. I think that the problem is that only a single module is unload from sys.modules.

If we want to save/restore modules, IMO the whole ``sys.modules`` dictionary must be saved/restored. Not a single entry.

Would it make sense to save/restore the whole sys.modules dict? Would it fix https://github.com/python/cpython/issues/108927 ?